src/tscore/Makefile.am

```diff
@@ -48,162 +48,74 @@ libtscore_la_LIBADD = \
 	-lc
 
 libtscore_la_SOURCES = \
-	Allocator.h \
 	AcidPtr.cc \
-	AcidPtr.h \
 	Arena.cc \
-	Arena.h \
 	ArgParser.cc \
-	ArgParser.h \
 	BaseLogFile.cc \
-	BaseLogFile.h \
-	BufferWriter.h \
-	BufferWriterForward.h \
 	BufferWriterFormat.cc \
 	ConsistentHash.cc \
-	ConsistentHash.h \
 	ContFlags.cc \
-	ContFlags.h \
 	CryptoHash.cc \
-	CryptoHash.h \
-	defalloc.h \
 	Diags.cc \
-	Diags.h \
 	EventNotify.cc \
-	EventNotify.h \
-	Extendible.h \
 	fastlz.c \
-	fastlz.h \
 	Hash.cc \
 	HashFNV.cc \
-	HashFNV.h \
-	Hash.h \
 	HashMD5.cc \
-	HashMD5.h \
 	HashSip.cc \
-	HashSip.h \
-	History.h \
 	HostLookup.cc \
-	HostLookup.h \
 	hugepages.cc \
-	hugepages.h \
-	I_Layout.h \
-	ink_aiocb.h \
-	ink_align.h \
-	ink_apidefs.h \
 	ink_args.cc \
-	ink_args.h \
 	ink_assert.cc \
-	ink_assert.h \
-	ink_atomic.h \
 	ink_base64.cc \
-	ink_base64.h \
 	ink_cap.cc \
-	ink_cap.h \
 	ink_code.cc \
-	ink_code.h \
 	ink_defs.cc \
-	ink_defs.h \
 	InkErrno.cc \
-	InkErrno.h \
 	ink_error.cc \
-	ink_error.h \
-	ink_exception.h \
 	ink_file.cc \
-	ink_file.h \
 	ink_hrtime.cc \
-	ink_hrtime.h \
 	ink_inet.cc \
-	ink_inet.h \
-	ink_inout.h \
-	ink_llqueue.h \
-	ink_lockfile.h \
-	INK_MD5.h \
 	ink_memory.cc \
-	ink_memory.h \
 	ink_mutex.cc \
-	ink_mutex.h \
-	ink_platform.h \
 	ink_queue.cc \
-	ink_queue.h \
 	ink_queue_utils.cc \
 	ink_rand.cc \
-	ink_rand.h \
 	ink_res_init.cc \
 	ink_res_mkquery.cc \
-	ink_resolver.h \
 	ink_resource.cc \
-	ink_resource.h \
 	ink_rwlock.cc \
-	ink_rwlock.h \
 	ink_sock.cc \
-	ink_sock.h \
 	ink_sprintf.cc \
-	ink_sprintf.h \
 	ink_stack_trace.cc \
-	ink_stack_trace.h \
 	ink_string.cc \
 	ink_string++.cc \
-	ink_string.h \
-	ink_string++.h \
 	ink_sys_control.cc \
-	ink_sys_control.h \
 	ink_syslog.cc \
-	ink_syslog.h \
 	ink_thread.cc \
-	ink_thread.h \
 	ink_time.cc \
-	ink_time.h \
 	ink_uuid.cc \
-	ink_uuid.h \
 	IpMap.cc \
 	IpMapConf.cc \
-	IpMapConf.h \
-	IpMap.h \
-	I_Version.h \
-	JeAllocator.h \
 	JeAllocator.cc \
 	Layout.cc \
-	List.h \
 	llqueue.cc \
 	lockfile.cc \
 	MatcherUtils.cc \
-	MatcherUtils.h \
-	MemSpan.h \
 	MemArena.cc \
-	MemArena.h \
 	MMH.cc \
-	MMH.h \
-	MT_hashtable.h \
 	ParseRules.cc \
-	ParseRules.h \
-	PriorityQueue.h \
-	Ptr.h \
 	RbTree.cc \
-	RbTree.h \
 	Regex.cc \
-	Regex.h \
 	Regression.cc \
-	Regression.h \
-	Result.h \
 	runroot.cc \
-	runroot.h \
 	signals.cc \
-	signals.h \
-	SimpleTokenizer.h \
 	SourceLocation.cc \
-	SourceLocation.h \
-	TestBox.h \
 	TextBuffer.cc \
-	TextBuffer.h \
 	Tokenizer.cc \
-	Tokenizer.h \
-	Trie.h \
-	TsBuffer.h \
```

Comment:
Yeh, I think this is fine. Having the include files in the SOURCES definitions is/was necessary (I think) when you used the default automakes rules to make source distribution (packages).  But we have our own Makefile target for that, so we shouldn't need to include the headers here.